boards/arduino-mkrzero: add initial support

aabadie opened the pull request:
And factorize code with arduino-mkr1000 since these boards are very similar.
The Arduino MKRZERO uses the second SPI to connect a SD card device, I configured it to use the one available in RIOT. This is not yet tested though.

I also noticed that there's a third one in the family: [Arduino MKRFOX1200](https://www.arduino.cc/en/Main.ArduinoBoardMKRFox1200) that offers a SigFox compatible antenna.

aabadie commented:
And this PR is based on #6666

aabadie commented:
since #6666 has been merged, I rebased this one. Tested mkr1000, still works.

aabadie commented on boards/arduino-mkrzero/include/board.h:
```diff
@@ -0,0 +1,72 @@
+ * @name    SD Card device configuration
+ * @{
+ */
+#define SDCARD_SPI_PARAM_SPI         (SPI_DEV(1))
```
This is the part, I'm not totally sure of. @haukepetersen or @vincent-d any comments?

haukepetersen replied:
Depends on what the question is... Does the `arduino-mkrzero` come with an on-board SD-card slot? If yes, then this looks valid to me (as long as the SD-card slot is connected to the pins mapped  for SPI_DEV(1)...).

aabadie replied:
It should be ok, regarding the [specs](https://www.arduino.cc/en/uploads/Main/ArduinoMKRZero-schematic.pdf). I just don't know what CD1 is on the SD card reader.

gebart replied:
Could be "card detect"?

gebart replied:
Could be "card detect"?

aabadie replied:
Thanks, googling points toward this direction. I added a define for this pin below in the file.

haukepetersen commented:
before merging this PR, I would however like to see a better solution for the `bossa` tool, as having the 4th board shipped with the same binary tool is ugly and bad -> so I would like to have a better solution before we go on. Sounds fair?

aabadie commented:
> Sounds fair?

Yes, I'm interested in looking into this except if you feel very motivated.

haukepetersen commented:
done: #7068

kbumsik commented on boards/arduino-mkr-common/include/periph_conf_common.h:
```diff
@@ -0,0 +1,270 @@
+        .clk_pin  = GPIO_PIN(PA, 13),
+        .miso_mux = GPIO_MUX_D,
+        .mosi_mux = GPIO_MUX_D,
+        .clk_mux  = GPIO_MUX_D,
```
I noticed my silly mistake while testing WINC1500 WiFi module. 🤕 Could you please change the MUX pads to:

```C
.miso_mux = GPIO_MUX_C,
.mosi_mux = GPIO_MUX_C,
.clk_mux  = GPIO_MUX_C,
```
Thank you.

aabadie replied:
no problem, directly squashed and pushed

aabadie commented on boards/arduino-mkr-common/include/periph_conf_common.h:
```diff
@@ -0,0 +1,270 @@
+    },
+    {
+        .dev      = &SERCOM2->SPI,
+        .miso_pin = GPIO_PIN(PA, 15),
```
This pin definition for the second SPI is different from mkr1000. I'll have to provide the SPI configuration for each boards.

haukepetersen commented:
now that #7068 is merged, would you mind to rebase and make use of it?

aabadie commented:
>  would you mind to rebase and make use of it?

I'm on it

aabadie commented:
@haukepetersen rebased and directly squashed

haukepetersen commented:
Looks good to me, though I have no board to test this... Anyone else?

aabadie commented:
anyone for testing this ?

aabadie commented:
@astralien3000, will you have time tomorrow for testing this one ? (I can lend you mine)

astralien3000 commented:
First, bossa does not compile on MacOS

aabadie commented:
>  bossa does not compile on MacOS

I think @kYc0o opened a PR about that some days ago.

kYc0o commented:
You can ACK my PR #7191 😃

aabadie commented:
@astralien3000, since the mac OS problem with is unrelated, can you try with a Linux VM ? At least, we can move forward here.

astralien3000 commented:
Sorry, but I can't flash it a all now (neither Mac nor Linux VM), I can't put it in flashing mode...

astralien3000 commented:
I tested `default` and `xtimer_usleep`, OK.
But `periph_timer` and `periph_pwm` don't seem to work (LED is oscillating, though)

aabadie commented:
are you on the most recent commit ?

astralien3000 commented:
hmm... I don't know, but since you did not add commits after my first post, I think I was.
I will check this on Monday (I don't have the computer I used for this at home).

aabadie commented:
Maybe there's a problem with PWM. Will check again when time permits. Thanks for testing !

astralien3000 commented:
I have checked, I was on the most recent commit. And I also needed to rebase to be able to flash the board (bossa problems with MacOS...).

aabadie commented:
@astralien3000, I pushed a fox regarding the PWM: pin ~2 and ~3 are working but I couldn't find the correct conf for pin ~4 and ~5. So there's only 2 available PWM pins in the default configuration.

Regarding the periph_timer test, I didn't test but it should work as I didn't change it and is working (normally) with mkr1000.

aabadie commented:
@astralien3000, if you have some time can you give another try to this one ?

astralien3000 commented:
- `periph_pwm` OK (for ~2 and ~3)
 - `periph_timer` still blocking

aabadie commented:
@astralien3000, thanks for testing again.

Regarding the timer test issue, I fixed it: by default mkrzero use the `sdcard_spi` module which itself uses xtimer and for some reason this messes up this test (TIMER_0 being initialized with different values apparently).
The fix is simply to remove the dependency to `sdcard_spi` which makes sense since this is up to the user to include it or not.

In the mean time, I also fixed the second spi periph configuration which was wrong.

Directly squashed and rebased.

Can you test again ?

astralien3000 commented:
Yes ! Now it works !

aabadie commented:
@haukepetersen, this one is ready now (and tested). Would be great if you could approve it (and merge ;)) !

aabadie commented:
@haukepetersen ?

haukepetersen commented on boards/arduino-mkr-common/include/arduino_pinmap.h:
```diff
@@ -56,7 +56,7 @@ extern "C" {
 #define ARDUINO_PIN_A5          GPIO_PIN(PA, 6)  /* AIN6 */
 #define ARDUINO_PIN_A6          GPIO_PIN(PA, 7)  /* AIN7 */
-/** @} */
+/** @ */
```
any reason for removing the `}`?

haukepetersen commented on boards/arduino-mkr-common/include/arduino_pinmap.h:
```diff
@@ -29,7 +29,7 @@ extern "C" {
 
 /**
- * @name    Mapping of MCU pins to Arduino pins
+ * @name   Mapping of MCU pins to Arduino pins
```
Indention off by one space.

haukepetersen commented on boards/arduino-mkr-common/include/board_common.h:
```diff
@@ -0,0 +1,51 @@
+ */
+#define XTIMER              TIMER_0
+#define XTIMER_CHAN         (0)
+/** @} */
```
please remove this block - no need to re-define the default configuration

haukepetersen commented on boards/arduino-mkr1000/include/board.h:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2016 Inria
+ * Copyright (C) 2017 Inria
```
Actually, I think convention is to use `2016-2017` when editing. Applies to many files of this PR

aabadie commented:
@haukepetersen, comments addressed.

haukepetersen commented:
all green -> go
